encoding/dictionary_based_int_iterator.go

```diff
@@ -0,0 +1,79 @@
+package encoding
+
+import (
+	"github.com/xichen2020/eventdb/generated/proto/encodingpb"
+	"github.com/xichen2020/eventdb/x/io"
+	"github.com/xichen2020/eventdb/x/proto"
+
+	bitstream "github.com/dgryski/go-bitstream"
+)
+
+// DictionaryBasedIntIterator iterates through a dict encoded stream of ints.
+type DictionaryBasedIntIterator struct {
+	bitReader               *bitstream.BitReader
```

Comment:
Should probably make this a field of the decoder, and call `bitReader.Reset(reader)` for every decode, and pass in the bitreader to the iterator constructor, so you don't need to create a new bit reader every time.

Reply:
👍